base/inference.jl

```diff
@@ -647,32 +642,38 @@ let stagedcache=Dict{Any,Any}()
             end
             f = ccall(:jl_instantiate_staged,Any,(Any,Any,Any),m,tt,env)
             stagedcache[(m,tt,env)] = f
-            return f.code
+            return f
         end
     end
 end
 
-function abstract_call_gf(f, fargs, argtype, e)
+function abstract_call_gf(f::ANY, fargs, argtype::ANY, e)
     argtypes = argtype.parameters
     tm = _topmod()
-    if length(argtypes)>1 && argtypes[2]===Int && (argtypes[1] <: Tuple ||
-       (isa(argtypes[1], DataType) && isdefined(Main, :Base) && isdefined(Main.Base, :Pair) &&
-        (argtypes[1]::DataType).name === Main.Base.Pair.name))
+    if length(argtypes)>2 && argtypes[3]===Int &&
+        (argtypes[2] <: Tuple ||
+         (isa(argtypes[2], DataType) && isdefined(Main, :Base) && isdefined(Main.Base, :Pair) &&
+          (argtypes[2]::DataType).name === Main.Base.Pair.name))
         # allow tuple indexing functions to take advantage of constant
         # index arguments.
         if istopfunction(tm, f, :getindex)
-            return getfield_tfunc(fargs, argtypes[1], argtypes[2])[1]
+            return getfield_tfunc(fargs, argtypes[2], argtypes[3])[1]
```

Comment:
Actually I realize that these are actually indices, so maybe not here, but there are other places.